sql/type.go

```diff
@@ -63,14 +63,15 @@ func (s Schema) CheckRow(row Row) error {
 	return nil
 }
 
-// Contains returns whether the schema contains a column with the given name.
-func (s Schema) Contains(column string) bool {
-	for _, col := range s {
-		if col.Name == column {
-			return true
+// Contains returns whether the schema contains a column with the given name
+// and the index of that column in the schema.
+func (s Schema) Contains(column string, source string) (int, bool) {
```

Comment:
Instead of making contains return an int, which feels weird, maybe we could have a `ColumnIndex` or `IndexOf` method that returns the index or `-1` when it's not found and the following `Contains`:

```go
func (s Schema) Contains(column string, source string) bool {
        return s.IndexOf(column, source) >= 0
}
```

WDYT?